### PR TITLE
Migrate to GitHub Actions for CI. Added the GitHub Action to build the SODA Dashboard.

### LINKED ISSUE #587: Enable CI with GitHub actions for the SODA Dashboard.
**Issue/Feature Description:**
The current CI across SODA Projects is being run using Travis CI. With the changes made to the free tier of Travis CI we are moving the CI for all the projects under SODA Foundation to GitHub Actions.
As part of this we need to move the CI for the dashboard project to GitHub Actions

**Why this issue to fixed / feature is needed(give scenarios or use cases):**
Currently the CI jobs are getting queued across the SODA Organisation and this is causing delays in the CI's being run if there are many PRs raised across projects. This delays PR reviews and merging. Moving to Github actions will run the CI independently.

**How to reproduce, in case of a bug:**

**Other Notes /  Environment Information: (Please give the env information, log link or any useful information for this issue)**

### PR DESCRIPTION
**Pull request by anvithks**
**What type of PR is this?**
 /kind enhancement

**What this PR does / why we need it**:
This PR enables GitHub actions to run the CI for the SODA Dashboard project.

**Which issue(s) this PR fixes**:
Fixes #587 

**Test Report Added?**:
/kind TESTED

**Test Report**:
Link to the test CI at my fork: [here](https://github.com/anvithks/opensds-dashboard/runs/2411475086?check_suite_focus=true)

The CI can be checked on this PR if GitHub Actions are enabled on the project.

**Special notes for your reviewer**:

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# This is a the GitHub Action to run the CI for SODA Dashboard
+
+name: SODA Dashboard CI
+
+# Controls when the action will run. 
+on:
+  # Triggers the workflow on push or pull request events but only for the main branch
+  push:
+    branches: [ master ]
```
Comment (asifdxtreme) [inline .github/workflows/ci.yml:9]: can we skip the branches clause, as we want our CI to run on all the branches whereever there is a change, this scenario will come into play when we have some dev or temp branch for development, at that point of time CI won't run on thoose branches.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+      
+  pull_request:
+    branches: [ master ]
```
Comment (asifdxtreme) [inline .github/workflows/ci.yml:12]: same thing as above

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # This workflow contains a single job called "build"
+  build:
+    # The type of runner that the job will run on
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      matrix:
+        os: [ubuntu-16.04]
```
Comment (asifdxtreme) [inline .github/workflows/ci.yml:26]: can we also add latest ubuntu OS? ( suggestion)

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        node-version: [8.x]
+        
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - uses: actions/checkout@v2
```
Comment (asifdxtreme) [inline .github/workflows/ci.yml:32]: this works but it's better give a name of the Step and then give `uses`  ( suggestion)

Comment (anvithks) [reply .github/workflows/ci.yml]: Fixed. 
I had to push multiple commits as there was some syntax error.

File: .github/workflows/ci.yml (hunk 3 above)
Comment (anvithks) [reply .github/workflows/ci.yml]: Added and updated the PR. @asifdxtreme

**Review by asifdxtreme — APPROVED** [review]
LGTM

**Review by skdwriting — APPROVED** [review]
LGTM

**Review by kumarashit — APPROVED** [review]

**Review by kumarashit — APPROVED** [review]
LGTM

**Comment by skdwriting** [comment]
CI migration in progress to GitHub actions, local ci ok. hence merging.

**Pull request merged**